Update for compatibility with puppet v5.5.2, closes #4

Tones29 opened the pull request:
When trying to run the puppet script on our new 18.04 server, I got the error below which seems to be related to the usage of an old puppet version (3.8.7). This PR updates the script so that it runs with the current puppet version. I have tested it on 16.04 and 18.04.

```
sudo puppet module install puppetlabs/apache
Traceback (most recent call last):
	17: from /usr/local/bin/puppet:23:in `<main>'
	16: from /usr/local/bin/puppet:23:in `load'
	15: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/bin/puppet:7:in `<top (required)>'
	14: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	13: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	12: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet/util/command_line.rb:12:in `<top (required)>'
	11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	10: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 9: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet.rb:8:in `<top (required)>'
	 8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 7: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 6: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet/util.rb:14:in `<top (required)>'
	 5: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet/util.rb:15:in `<module:Puppet>'
	 4: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet/util.rb:16:in `<module:Util>'
	 3: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 1: from /var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet/util/monkey_patches.rb:172:in `<top (required)>'
/var/lib/gems/2.5.0/gems/puppet-3.8.7/lib/puppet/util/monkey_patches.rb:178:in `<class:SSLContext>': undefined method `<<' for nil:NilClass (NoMethodError)
```

Tones29 committed:
Update for compatibility with puppet v5.5.2, closes #4

tfoote commented:
Thanks for taking the time to update it. Do you believe that this is backwards compatible with puppet 3? And what is the traceback related to?

Tones29 commented:
The traceback above is related to the error I got when trying to install a new rosmirror with the current revision f4110d1 on ubuntu 18.04.

For backwards compatibility to existing installations with puppet 3.8.7 we would have to add `/etc/puppet/modules` to the modulepath. Then it works at least on our old ROS mirror.

tfoote commented:
Great, could you add that back so we have both paths and that will mean that we don't need to fork development or force maintainers to upgrade. But we can give them the opportunity to upgrade when convenient.

Tones29 committed:
Add modulepath for compatibility with puppet 3.x

This modulepath can be removed as soon as no rosmirror uses puppet 3.x any more.

Tones29 commented:
That makes sense, I just added the line. I think there might be some rosmirrors which use puppet 3.x and run an ubuntu version >12.04, but I am not sure. We should probably check with all maintainers before removing the modulepath.

tfoote reviewed (approved):
Thanks for the iteration I was able to test it on some legacy platforms successfully.

ros-discourse commented:
This pull request has been mentioned on **ROS Discourse**. There might be relevant details there:

https://discourse.ros.org/t/updated-mirror-list-available/14250/1
